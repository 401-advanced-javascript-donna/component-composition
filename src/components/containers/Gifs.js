import React, { Component } from 'react';
import Deck from '../Deck';
import PropTypes from 'prop-types';
import apiCall from '../../services/api-call';

class Gifs extends Component {
  static propTypes = {
    gifsCount: PropTypes.number.isRequired
  }
  state= {
    gifs: [],
  }

  componentDidMount() {
    apiCall('https://hey-arnold-api.herokuapp.com/api/v1/gifs', this.props.gifsCount)
      .then(result => this.setState({ gifs: result }));
  }

  generateImages = () => {
    apiCall('https://hey-arnold-api.herokuapp.com/api/v1/gifs', this.props.gifsCount)
      .then(result => this.setState({ gifs: result }));
  }

  render() {
    let images = this.state.gifs.map(gif => {
      return <img key={gif._id} src={gif.gifLink} />;
    });
    return (
      <>
        <Deck title='gifs'>
          {images}
        </Deck>
        <button onClick={this.generateImages}>Get New Gifs</button>
      </>
    );
  }
}

export default Gifs;

