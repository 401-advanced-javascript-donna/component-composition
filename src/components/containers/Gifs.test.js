import React from 'react';
import { shallow } from 'enzyme';
import Gifs from './Gifs';

describe('Gifs container', () => {
  global.fetch = jest.fn(() => Promise.resolve({}));
  it('renders Gifs', () => {
    const wrapper = shallow(<Gifs gifsCount={20} />);
    expect(wrapper).toMatchSnapshot();
  });
});
