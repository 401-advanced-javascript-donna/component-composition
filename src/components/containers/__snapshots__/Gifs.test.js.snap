// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gifs container renders Gifs 1`] = `
<Fragment>
  <Deck
    title="gifs"
  />
  <button
    onClick={[Function]}
  >
    Get New Gifs
  </button>
</Fragment>
`;
